batchflow/tests/pipeline_init_model_test.py

```diff
@@ -0,0 +1,55 @@
+""" Tests for pipeline init_model action """
+# pylint: disable=import-error, no-name-in-module
+import pytest 
+
+from batchflow import DatasetIndex, Pipeline, Batch, C
+from batchflow.models import BaseModel
+from batchflow.model_dir import NonInitializedModel
+
+@pytest.fixture
+def dataset():
+    """ Toy dataset """
+    index = DatasetIndex(100)
+    return Dataset(index, Batch)
+
+@pytest.mark.parametrize('use_algebra', [True, False])
+@pytest.mark.parametrize('config', [dict(mode='static', base_class=BaseModel, name='model', config={}),
+                                    dict(mode='dynamic', base_class=BaseModel, name='model', config={})])
+def test_single_model(use_algebra, config, dataset):
+    """ Verifies that the model in init_model is initialized as expected with config provided, i.e.
```

Comment:
This test does not test anything. The model is never addressed and never used in training or inference.

Reply:
I made a bit messy. We are not interested here in addressing the model but storing

Reply:
> This test does not test anything. The model is never addressed and never used in training or inference.

I made a bit messy. We are not interested here in addressing the model but storing

Reply:
There is no point in storing anything if you never address it in a proper way. You can store it in /dev/null then.

Moreover, you killed the whole idea of static models without even realizing it.

Furthermore, `pipeline.init_model` and `pipeline.before.init_model` now works differently.

Besides, you evaluate named expressions in a peculiar fashion which is also very different from the rest of `Pipeline`.